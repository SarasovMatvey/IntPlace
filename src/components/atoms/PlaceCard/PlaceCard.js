import { Card, Image } from "semantic-ui-react";
import noImage from "./img/no-image.png";

function PlaceCard({
  poster,
  title,
  description,
  className,
  fluid,
  onClick,
  placeId,
  children,
}) {
  return (
    <Card
      data-id={placeId}
      className={className || ""}
      fluid={fluid}
      onClick={onClick}
    >
      <Image src={poster || noImage} wrapped ui={false} />
      <Card.Content>
        <Card.Header>{title}</Card.Header>
        <Card.Description>{description}</Card.Description>
        {children}
      </Card.Content>
    </Card>
  );
}

export default PlaceCard;
