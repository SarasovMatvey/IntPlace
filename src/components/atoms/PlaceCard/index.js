export { default } from "./PlaceCard.js";
