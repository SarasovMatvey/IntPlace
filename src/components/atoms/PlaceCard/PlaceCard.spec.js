import PlaceCard from "./PlaceCard";

const setUp = props => shallow(<PlaceCard {...props} />);

describe("PlaceCard", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it("should renders properly", () => {
    expect(component).toMatchSnapshot();
  });
});
