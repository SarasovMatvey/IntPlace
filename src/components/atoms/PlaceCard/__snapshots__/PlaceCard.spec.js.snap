// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard should renders properly 1`] = `
<Card
  className=""
>
  <Image
    as="img"
    src="no-image.png"
    ui={false}
    wrapped={true}
  />
  <CardContent>
    <CardHeader />
    <CardDescription />
  </CardContent>
</Card>
`;
